core/services/pipeline/orm.go

```diff
@@ -202,33 +226,40 @@ func (o *orm) InsertFinishedRun(db *gorm.DB, run Run, trrs []TaskRunResult, save
 	if run.Outputs.Val == nil || len(run.Errors) == 0 {
 		return 0, errors.Errorf("run must have both Outputs and Errors, got Outputs: %#v, Errors: %#v", run.Outputs.Val, run.Errors)
 	}
-	if len(trrs) == 0 && (saveSuccessfulTaskRuns || run.HasErrors()) {
+	if len(run.PipelineTaskRuns) == 0 && (saveSuccessfulTaskRuns || run.HasErrors()) {
 		return 0, errors.New("must provide task run results")
 	}
 
-	err = postgres.GormTransactionWithoutContext(db, func(tx *gorm.DB) error {
-		if err = tx.Create(&run).Error; err != nil {
+	ctx, cancel := postgres.DefaultQueryCtx()
+	defer cancel()
+	err = postgres.SqlxTransaction(ctx, db, func(tx *sqlx.Tx) error {
+		sql := `INSERT INTO pipeline_runs (pipeline_spec_id, meta, errors, inputs, outputs, created_at, finished_at, state)
+		VALUES (:pipeline_spec_id, :meta, :errors, :inputs, :outputs, :created_at, :finished_at, :state)
+		RETURNING *;`
+
+		query, args, e := tx.BindNamed(sql, run)
+		if e != nil {
```

Comment:
should be `err` here and in previous line?

Reply:
This is `e` to avoid lint issues because `err` is a named param. If I change it to `err` then I need to use `=` and declare `query` and `args` via a var declaration beforehand.